legal-api/src/legal_api/resources/v2/namerequest.py

```diff
@@ -40,3 +41,40 @@ def get(identifier):
         current_app.logger.error(err)
         abort(500)
         return {}, 500  # to appease the linter
+
+
+@bp.route('/<string:identifier>/validate', methods=['GET'])
+@cross_origin(origin='*')
+def validate_with_contact_info(identifier):
+    """Return a JSON object with name request information."""
+    try:
+        # The request must include email or phone number
+        email = request.args.get('email', None)
+        phone = request.args.get('phone', None)
+        if not (email or phone):
+            return make_response(jsonify(message='The request must include email or phone number.'), 403)
+
+        nr_response = namex.query_nr_number(identifier)
+        # Errors in general will just pass though,
+        # 404 is overriden as it is giving namex-api specific messaging
+        if nr_response.status_code == 404:
+            return make_response(jsonify(message='{} not found.'.format(identifier)), 404)
+
+        nr_json = nr_response.json()
+
+        # Check the NR is affiliated with this account
+        orgs_response = AccountService.get_account_by_affiliated_identifier(identifier)
+        if len(orgs_response['orgs']):
+            return jsonify(nr_json)
+
+        # If NR is not affiliated, validate the email and phone
+        nr_phone = nr_json.get('applicants').get('phoneNumber')
+        nr_email = nr_json.get('applicants').get('emailAddress')
+        if (phone and phone != nr_phone) or (email and email != nr_email):
+            return make_response(jsonify(message='Invalid email or phone number.'), 400)
```

Comment:
is this logic correct?  we only need one of email or phone to match the NR applicant info returned right?

![image](https://github.com/bcgov/lear/assets/6685223/96f103f3-b01c-4535-ba1f-9f63aebafbd8)

Reply:
Yeah, I thought about that, too.

The code as written will, I believe, validate one or the other or both if both are provided. I think that makes sense (I mean, you wouldn't provided the data if email matched but not phone, right?).

Reply:
true, ok i'm good then.